src/transformers/tokenization_utils_base.py

```diff
@@ -1903,7 +1964,34 @@ def tojson(x, ensure_ascii=False, indent=None, separators=None, sort_keys=False)
             # We also expose some options like custom indents and separators
             return json.dumps(x, ensure_ascii=ensure_ascii, indent=indent, separators=separators, sort_keys=sort_keys)
 
-        jinja_env = ImmutableSandboxedEnvironment(trim_blocks=True, lstrip_blocks=True)
+        class AssistantTracker(Extension):
+            tags = {"generation"}
+
+            def __init__(self, environment):
+                super().__init__(environment)
+                environment.extend(new_generation_trackers=self.new_generation_trackers)
```

Comment:
What type is `environment` here? I'm not very familiar with jinja, so it might be not knowing the standard patterns here, but it seems messy to accept and modify the env state, but not have it as part of the instance state